platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -386,27 +387,29 @@ class SFP(SfpBase):
     SFP_MLNX_ERROR_BIT_PCIE_POWER_SLOT_EXCEEDED = 0x00080000
     SFP_MLNX_ERROR_BIT_RESERVED = 0x80000000
 
-    def __init__(self, sfp_index, slot_id=0, linecard_port_count=0, lc_name=None):
+    def __init__(self, sfp_index, sfp_type=None, slot_id=0, linecard_port_count=0, lc_name=None):
         super(SFP, self).__init__()
         
+        self._sfp_type = sfp_type
         if slot_id == 0: # For non-modular chassis
             self.index = sfp_index + 1
             self.sdk_index = sfp_index
 
-            from .thermal import initialize_sfp_thermal
-            self._thermal_list = initialize_sfp_thermal(sfp_index)
+            if self._sfp_type != RJ45_TYPE:
```

Comment:
i think the code should have more comments to understand why this path was taken. why thermal is not relevant fro rj45?